src/models/data.py

```diff
@@ -170,10 +183,21 @@ def __len__(self) -> int:
     def _load_datasets(data_path: Path, mode: str,
                        shuffle_data: bool, experiment: str,
                        mask: Optional[List[bool]] = None,
-                       pred_months: Optional[List[int]] = None) -> List[Path]:
+                       pred_months: Optional[List[int]] = None,
+                       test_year: Optional[int] = None,
+                       test_month: Optional[int] = None) -> List[Path]:
 
-        data_folder = data_path / f'features/{experiment}/{mode}'
+        if (test_year is None) and (test_month is None):
```

Comment:
(I see now that this is handled in the base training class, but since the `DataLoader` is something we would want to expose to users, I think it makes sense to clean up here)

Reply:
can we chat through how to improve this API ? So far I have the ideas:

1) Have a function to check whether the file exists in the train datafolder 
1) Expose this to the user from the `DATALOADER` not the base `neural_network` class
1) Require making predictions of whole years (don't expose months to users)

Did i miss any?

Reply:
for 3., would it be possible to expose both, but if `test_month` is `None`, then do the whole year?